gauntlet/packages/gauntlet-solana/src/commands/internal/solana.ts

```diff
@@ -124,20 +357,24 @@ export default abstract class SolanaCommand extends WriteCommand<TransactionResp
 
   simulateTx = async (signer: PublicKey, txInstructions: TransactionInstruction[], feePayer?: PublicKey) => {
     try {
-      const { blockhash, lastValidBlockHeight } = await this.provider.connection.getLatestBlockhash()
-      const tx = makeTx(txInstructions, {
-        feePayer: feePayer || signer,
-        blockhash,
-        lastValidBlockHeight,
+      const { blockhash } = await this.provider.connection.getLatestBlockhash()
+      // TODO: accept a tx pre-made tx without the signatures
+      const tx = makeTx({
+        instructions: txInstructions,
+        recentBlockhash: blockhash,
+        payerKey: feePayer || signer,
       })
       // simulating through connection allows to skip signing tx (useful when using Ledger device)
-      const { value: simulationResponse } = await this.provider.connection.simulateTransaction(tx)
+      const { value: simulationResponse } = await this.provider.connection.simulateTransaction(tx, {
+        commitment: CONFIRM_LEVEL,
+      })
       if (simulationResponse.err) {
         throw new Error(JSON.stringify({ error: simulationResponse.err, logs: simulationResponse.logs }))
       }
       logger.success(`Tx simulation succeeded: ${simulationResponse.unitsConsumed} units consumed.`)
       return simulationResponse.unitsConsumed
     } catch (e) {
+      console.log(e.message)
```

Comment:
This should be handled upstream, after we throw

Reply:
I had to log here since some of the errors returned weren't parsable JSON and the error handling code would crash